homeassistant/components/light/decora_wifi.py

```diff
@@ -60,8 +61,13 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
         perms = session.user.get_residential_permissions()
         all_switches = []
         for permission in perms:
-            acct = ResidentialAccount(session, permission.residentialAccountId)
-            for residence in acct.get_residences():
+            if permission.residentialAccountId is not None:
+                acct = ResidentialAccount(session, permission.residentialAccountId)
+                for residence in acct.get_residences():
```

Comment:
We could store these in a list and potentially add the single residence below before we iterate the switches of all residences.